### PR TITLE
Add test case for measurements with no output configuration in client generation acceptance test

### PR DESCRIPTION
**Pull request by Jotheeswaran-Nandagopal**
- [x] This contribution adheres to [CONTRIBUTING.md](https://github.com/ni/measurement-services-python/blob/main/CONTRIBUTING.md).

### What does this Pull Request accomplish?

- Added a measurement with no output configurations to test utilities.
- Added a test case for measurements with no output configuration in `test_client_generator.py`.
- Added a if condition in `mako`, to remove missing imports during the generation of measurements with no output configuration.
- Remove duplicate imports in `mako`.

### Why should this Pull Request be merged?

- Added a new test case to verify the client generation for the measurements with no output configuration.
- Handle unused imports in `mako`. As, some imports are unused in the client module for the measurements with no output configuration.
- Remove duplicate imports i.e., Path is imported twice, as `pathlib` is already added in `mako`, also, we are adding `pathlib`, during client creation of measurement with path configurations.

### What testing has been done?

- Existing and new automation tests passed.

### CODE REVIEW

File: packages/generator/tests/test_assets/example_renders/measurement_plugin_client/void_measurement_client.py [hunk 1]
```diff
@@ -0,0 +1,221 @@
...
+            input_metadata=list(self._configuration_metadata.values()),
+            output_metadata=list(self._output_metadata.values()),
+            service_name=self._service_class,
+            pool=descriptor_pool.Default(),
+        )
+
+    def _create_measure_request(
+        self, parameter_values: List[Any]
+    ) -> v2_measurement_service_pb2.MeasureRequest:
+        serialized_configuration = any_pb2.Any(
```
Comment (Avinash2Suresh) [inline packages/generator/tests/test_assets/example_renders/measurement_plugin_client/void_measurement_client.py:156]: we would need to add type url here once the other PR for [C# service support](https://github.com/ni/measurement-plugin-python/pull/889/) goes in.

File: packages/generator/ni_measurement_plugin_sdk_generator/client/templates/measurement_plugin_client.py.mako [hunk 2]
```diff
@@ -3,19 +3,15 @@
...
 import threading
 % if len(enum_by_class_name):
 from enum import Enum
 % endif
-% for module in built_in_import_modules:
-${module}
-% endfor
-from typing import Any, Generator, Iterable, List, NamedTuple, Optional
+from pathlib import Path
+from typing import Any, Generator${", Iterable" if output_metadata else ""}, List${", NamedTuple" if output_metadata else ""}, Optional
```
Comment (bkeryan) [inline packages/generator/ni_measurement_plugin_sdk_generator/client/templates/measurement_plugin_client.py.mako:11]: Let's use a Python block to make this easier to maintain.
```suggestion
<%
    typing_imports = ["Any", "Generator", "List", "Optional"]
    if output_metadata:
        typing_imports += ["Iterable", "NamedTuple"]
%>\
from typing import ${", ".join(sorted(typing_imports))}
```

File: packages/generator/tests/acceptance/test_client_generator.py [hunk 3]
```diff
@@ -9,14 +9,18 @@
...
-from tests.utilities.measurements import non_streaming_data_measurement, streaming_data_measurement
+from tests.utilities.measurements import (
+    non_streaming_data_measurement,
+    streaming_data_measurement,
+    void_measurement,
+)
 
 
-def test___command_line_args___create_client___render_without_error(
+def test___command_line_args_for_non_streaming_measurement___create_client___render_without_error(
```
Comment (bkeryan) [inline packages/generator/tests/acceptance/test_client_generator.py:19]: If every precondition says "command line args" then you can remove it from all of them. Just describe what's different.

- non_streaming_measurement
- void_measurement
- all_registered_measurements
- interactive_mode
- etc.

**Review by bkeryan — CHANGES_REQUESTED** [review]

File: packages/generator/tests/acceptance/test_client_generator.py (hunk 3 above)
Comment (Jotheeswaran-Nandagopal) [reply packages/generator/tests/acceptance/test_client_generator.py]: I have updated the names of the tests by removing `command_line_args`.

**Review by bkeryan — APPROVED** [review]

**Pull request merged**